Contributing.md

```diff
@@ -24,33 +24,55 @@ This guide is for developers who wish to contribute to the Dolphin codebase. It
 
+This project uses clang-format (stable branch) to check for common style issues. In case of conflicts between this guide and clang-format rules, the latter should be followed instead of this guide.
+
+
+### Checking and fixing formatting issues
+
+In most cases, clang-format can and **should** be used to automatically reformat code and solve most formatting issues.
+
+- To run clang-format on all staged files:
+  ```
+  git diff --cached --name-only | egrep '[.](cpp|h|mm)$' | xargs clang-format -style=file -i
+  ```
+
+- Formatting issues can be checked for before committing with a lint script that is included with the codebase. To enable it as a pre-commit hook (assuming you are in the repository root):
+  ```
+  ln -s ../../Tools/lint.sh .git/hooks/pre-commit
+  ```
+
+  To also automatically fix common formatting issues, use the reformat.sh hook instead:
+  ```
+  ln -sf ../../Tools/reformat.sh .git/hooks/pre-commit
+  ```
+  Note that the hook will completely reformat all staged files and commit them, so avoid using git commit -p or add -p or similar when using this hook.
+
+
 ## Styling and formatting
 
 ### General
 - Try to limit lines of code to a maximum of 100 characters.
     - Note that this does not mean you should try and use all 100 characters every time you have the chance. Typically with well formatted code, you normally shouldn't hit a line count of anything over 80 or 90 characters.
-- The indentation style we use is tabs for initial indentation and then, if vertical alignment is needed, spaces are to be used:
+- The indentation style we use is 2 spaces per level:
 ```c++
 class IndentAndAlignmentSample
 {
 public:
-	void ThisMethodIsIndentedByOneLevel(int using_one_single_tab)
-	{
-		// this method, along with its opening and closing braces are
-		// indented with a single tab. This comment however is indented
-		// with two tabs. There is no vertical alignment yet, so no
-		// spaces are involved at this point.
-		m_other->DoStuff(m_first_parameter,
-		                 m_second_parameter,
-		                 m_third_parameter);
-		// Indent for the three previous lines is done using two tabs
-		// each (which brings the lines to the column where the word
-		// m_other begins in the first line).
-		// However, lines two and three are vertically aligned using
-		// 17 spaces (that's the length of "m_other->DoStuff(") in order
-		// to line up the method parameters correctly, regardless of
-		// tab-width settings used in your editor/IDE.
-	}
+  void ThisMethodIsIndentedByOneLevel(int using_two_spaces)
+  {
+    // this method, along with its opening and closing braces are
+    // indented with two spaces. This comment however is indented
+    // with four spaces.
+    m_other->DoStuff(m_first_parameter,
+                     m_second_parameter,
+                     m_third_parameter);
```

Comment:
Please make a note to only do this kind of wrapping when it's necessary because the line would be too long otherwise. Doing what this example actually shows wouldn't be appropriate.

Alternatively (and probably preferably), you could remove the entire example. It was added to better explain the combination of tabs and spaces that we used before, because people got a bit confused when we only had text without an example. "Two spaces" is much easier to explain.